### PR TITLE
84 switch server containers to map

### PR DESCRIPTION
**Pull request by forbidden-arts**
switched to map. rewrote a lot of the configParser because it relied on vector specific stuff (like .back). passes all the thunder tests except one of the returns is differend (that matches previous branches, so it's outside of the scope of this PR).

### CODE REVIEW

File: config/bad/bad-dup-names.conf [hunk 1]
```diff
@@ -0,0 +1,98 @@
+
+server
+{
+	main
+	{
+		host 127.0.0.1
+		listen 8000
+		server_name localhost
```
Comment (merituulie) [inline config/bad/bad-dup-names.conf:8]: we should use some other name than the localhost, its a convention and reserved by default already to the ip 127.0.0.1 ...

File: public_www/upload/.gitkeep
Comment (merituulie) [inline public_www/upload/.gitkeep:1]: who dis?

Comment (forbidden-arts) [reply public_www/upload/.gitkeep:1]: git doesn't keep empty directories, but we need this one to pass parsing.

File: config/bad/bad-dup-names.conf (hunk 1 above)
Comment (forbidden-arts) [reply config/bad/bad-dup-names.conf:8]: that's fine. i tried a bunch of options, and that was the last one. i'll update it later

**Pull request closed without merging**